### ISSUE TITLE
Grouped repo PACKAGES.gz is different from proxy repo

### ISSUE THREAD

**Issue by cderv** [sonatype-nexus-community/nexus-repository-r#39]
I found a strange behaviour when trying to understand how everything work. It seems that PACKAGES.gz file are different from proxy repo and group repo. I expected them to be the same, without considering hosted repo. Basically group = proxy + hosted.

Here, some example you can reproduce by modifying the url

My repo url

``` r
nexus_repos <- list(Group = "http://10.132.12.126:8081/repository/DDL-R-Group/", 
                    Proxy = "http://10.132.12.126:8081/repository/DDL-R-Proxy/", 
                    publish = "http://10.132.12.126:8081/repository/DDL-R-publish")
```

Get the metadata about what is available through the repos

``` r
pkgs_group <- as.data.frame(available.packages(repos = nexus_repos$Group, type = "source"))
pkgs_proxy <- as.data.frame(available.packages(repos = nexus_repos$Proxy, type = "source"))
```

As a use case to show the difference, let's take `installr` that is a windows package only. It has a field `OS_type = windows`.  See on [CRAN](https://cran.rstudio.com/web/packages/installr).  Thus, it should be filtered out by default filters because I am on linux. (see ?available.packages for details). 
However, it is not because this field is missing

``` r
subset(pkgs_group, Package == "installr")
#>           Package Version Priority                       Depends Imports
#> installr installr  0.20.0     <NA> R (>= 2.14.0), stringr, utils    <NA>
#>          LinkingTo
#> installr      <NA>
#>                                                                            Suggests
#> installr curl, XML, htmltab, devtools, rjson, data.table, plyr,\nggplot2, sp, tools
#>          Enhances License License_is_FOSS License_restricts_use OS_type
#> installr     <NA>   GPL-2            <NA>                  <NA>    <NA>
#>          Archs MD5sum NeedsCompilation File
#> installr  <NA>   <NA>               no <NA>
#>                                                            Repository
#> installr http://10.132.12.126:8081/repository/DDL-R-Group/src/contrib
```

On the contratry, this is working for proxy repo

``` r
subset(pkgs_proxy, Package == "installr")
#>  [1] Package               Version               Priority             
#>  [4] Depends               Imports               LinkingTo            
#>  [7] Suggests              Enhances              License              
#> [10] License_is_FOSS       License_restricts_use OS_type              
#> [13] Archs                 MD5sum                NeedsCompilation     
#> [16] File                  Repository           
#> <0 rows> (or 0-length row.names)
```

because the field is present in PACKAGE.gz for proxy repo.

``` r
temp_file <- tempfile(fileext = ".gz")
download.file("http://10.132.12.126:8081/repository/DDL-R-Proxy/src/contrib/PACKAGES.gz", destfile = temp_file, mode = "wb")
pkgs_proxy_dcf <- as.data.frame(read.dcf(temp_file))
subset(pkgs_proxy_dcf, Package == "installr")
#>       Package Version                       Depends
#> 5222 installr  0.20.0 R (>= 2.14.0), stringr, utils
#>                                                                        Suggests
#> 5222 curl, XML, htmltab, devtools, rjson, data.table, plyr,\nggplot2, sp, tools
#>      License NeedsCompilation Imports LinkingTo Enhances
#> 5222   GPL-2               no    <NA>      <NA>     <NA>
#>      License_restricts_use OS_type Priority License_is_FOSS Archs MD5sum
#> 5222                  <NA> windows     <NA>            <NA>  <NA>   <NA>
#>      Path
#> 5222 <NA>
```

If I do not apply the filter, it stays

``` r
pkgs_proxy <- as.data.frame(available.packages(repos = nexus_repos$Proxy, type = "source", 
                                               # leaving out OS_type
                                               filters = c("R_version", "subarch", "duplicates")))
subset(pkgs_proxy_dcf, Package == "installr")
#>       Package Version                       Depends
#> 5222 installr  0.20.0 R (>= 2.14.0), stringr, utils
#>                                                                        Suggests
#> 5222 curl, XML, htmltab, devtools, rjson, data.table, plyr,\nggplot2, sp, tools
#>      License NeedsCompilation Imports LinkingTo Enhances
#> 5222   GPL-2               no    <NA>      <NA>     <NA>
#>      License_restricts_use OS_type Priority License_is_FOSS Archs MD5sum
#> 5222                  <NA> windows     <NA>            <NA>  <NA>   <NA>
#>      Path
#> 5222 <NA>
```

I don't know what is missing for hosted repo as it is difficult for me to understand Java code. However, I can help with R knowledge on all this. Metadata are really important for a repo to work with R functions. 
I'll continue to test (with devtools for example) and report other behaviors

**Comment by Alexandor84** [comment]
I think I have a similar problem. There are two repos: **proxy** (remote storage https://cloud.r-project.org) and **group**.

List of packages are different, please see example below:

install.packages('shiny',  repos='https://localhost/nexus/repository/test-r-group/')
Installing package into ‘/usr/local/lib/R/site-library’
also installing the dependencies ‘Rcpp’, ‘httpuv’, ‘mime’, ‘jsonlite’, ‘xtable’, ‘digest’, ‘htmltools’, ‘R6’, ‘sourcetools’, ‘later’, ‘promises’, ‘crayon’, ‘rlang’

install.packages('shiny',  repos='https://localhost/nexus/repository/test-r-proxy/')
Installing package into ‘/usr/local/lib/R/site-library’
also installing the dependencies ‘Rcpp’, **‘BH’**, ‘httpuv’, ‘mime’, ‘jsonlite’, ‘xtable’, ‘digest’, ‘htmltools’, ‘R6’, ‘sourcetools’, ‘later’, ‘promises’, ‘crayon’, ‘rlang’

The "**BH**" dependency is missing if I use **group** repo.

Does anyone have an idea how to fix it?

**Comment by theWizK** [comment]
I'm not sure this is the entire problem with why grouped repo PACKAGES.gz is different from the proxied repo, but ONE of the problems is definitely that the LinkingTo field is missing from the headers in the grouped repo's version. I have fixed this in a forked repository and will submit a pull request for it shortly...

**Comment by DarthHater** [comment]
@theWizK you rule! This is why we do Open Source :)

**Comment by theWizK** [comment]
FYI -- the pull request was completed.

**Comment by aornatovskyy** [comment]
Will be fixed in the Nexus Repository Manager 3.20.0

**Comment by aornatovskyy** [comment]
Fixed in the NXRM 3.20.0-04 - R 1.1.1
